src/cdk/drag-drop/drop-list-ref.ts

```diff
@@ -367,11 +367,20 @@ export class DropListRef<T = any> {
    * @param items Items that are a part of this list.
    */
   withItems(items: DragRef[]): this {
+    const previousItems = this._draggables;
     this._draggables = items;
     items.forEach(item => item._withDropContainer(this));
 
     if (this.isDragging()) {
-      this._cacheItems();
+      const draggedItems = previousItems.filter(item => item.isDragging());
+
+      // If all of the items being dragged were removed
+      // from the list, abort the current drag sequence.
+      if (draggedItems.every(item => items.indexOf(item) === -1)) {
```

Comment:
won't this still have an issue if only some of the items have been destroyed?

Reply:
The problem here is that nothing ever tells the list that dragging has stopped. If only part of the dragged items are destroyed mid-drag, there should be at least one left to notify the list once it's dropped.

Reply:
Ah ok, I misunderstood. I thought the issue was with trying to attach items that had been destroyed